### PR TITLE
Add readonly type modifier to compile arguments

### PR DESCRIPTION
**Pull request by NickKoester**
Please fill in this template.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by NickKoester** [comment]
This addresses one of the points in the common mistakes section about using readonly arrays when the function doesn't modify it at all. This would allow for better type narrowing when providing an array of keywords to the compile function

**Comment by deltaidea** [comment]
Is pinning the TS version to 3.4 going to break anything for others? Does Moo itself support TS <3.4?

File: types/moo/index.d.ts [hunk 1]
```diff
@@ -70,7 +72,7 @@ export interface Rule {
     type?: TypeMapper;
 }
 export interface Rules {
-    [x: string]: RegExp | string | string[] | Rule | Rule[] | ErrorRule | FallbackRule;
+    [x: string]: RegExp | string | readonly string[] | Rule | Rule[] | ErrorRule | FallbackRule;
```
Comment (amcasey) [inline types/moo/index.d.ts]: Why isn't the Rule array also readonly?

**Review by amcasey — CHANGES_REQUESTED** [review]
Personally, I wouldn't bump the TS version just for this, but I'll defer to the owners.

**Pull request closed without merging**